### ISSUE TITLE
Install fails with DockerHub builds

### ISSUE THREAD

**Issue by armandmcqueen** [horovod/horovod#1079]
I am trying to build [this](https://github.com/armandmcqueen/tensorpack-mask-rcnn/blob/master/Dockerfile) Dockerfile using DockerHub builds. I have no problems building locally on Mac or on an EC2 instance, but DockerHub builds consistently fail at the install Horovod step (see below).

Can you help point me in the right direction?  

* Is the 'error code -4' something I can investigate?
* Has anyone else tried setting up DockerHub autobuilds with Horovod?

```
Step 6/15 : RUN ldconfig /usr/local/cuda-10.0/targets/x86_64-linux/lib/stubs && HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_WITH_TENSORFLOW=1 pip install --no-cache-dir horovod==0.15.2 && ldconfig
---> Running in 6a964b96d01b
Collecting horovod==0.15.2
Downloading https://files.pythonhosted.org/packages/17/1a/3e836e3f659d3b331afcb679f84c46c8d489c2b8c21b1d85a99ac31d3480/horovod-0.15.2.tar.gz (119kB)
Requirement already satisfied: cffi>=1.4.0 in /usr/local/lib/python3.6/site-packages (from horovod==0.15.2) (1.12.2)
Requirement already satisfied: pycparser in /usr/local/lib/python3.6/site-packages (from cffi>=1.4.0->horovod==0.15.2) (2.19)
Installing collected packages: horovod
Running setup.py install for horovod: started
Running setup.py install for horovod: finished with status 'error'
Complete output from command /usr/local/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-0x3phzja/horovod/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-dm5gilyr/install-record.txt --single-version-externally-managed --compile:
running install
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.6
creating build/lib.linux-x86_64-3.6/horovod
copying horovod/__init__.py -> build/lib.linux-x86_64-3.6/horovod
creating build/lib.linux-x86_64-3.6/horovod/keras
copying horovod/keras/__init__.py -> build/lib.linux-x86_64-3.6/horovod/keras
copying horovod/keras/callbacks.py -> build/lib.linux-x86_64-3.6/horovod/keras
creating build/lib.linux-x86_64-3.6/horovod/tensorflow
copying horovod/tensorflow/compression.py -> build/lib.linux-x86_64-3.6/horovod/tensorflow
copying horovod/tensorflow/mpi_ops.py -> build/lib.linux-x86_64-3.6/horovod/tensorflow
copying horovod/tensorflow/__init__.py -> build/lib.linux-x86_64-3.6/horovod/tensorflow
creating build/lib.linux-x86_64-3.6/horovod/torch
copying horovod/torch/compression.py -> build/lib.linux-x86_64-3.6/horovod/torch
copying horovod/torch/mpi_ops.py -> build/lib.linux-x86_64-3.6/horovod/torch
copying horovod/torch/__init__.py -> build/lib.linux-x86_64-3.6/horovod/torch
creating build/lib.linux-x86_64-3.6/horovod/_keras
copying horovod/_keras/__init__.py -> build/lib.linux-x86_64-3.6/horovod/_keras
copying horovod/_keras/callbacks.py -> build/lib.linux-x86_64-3.6/horovod/_keras
creating build/lib.linux-x86_64-3.6/horovod/common
copying horovod/common/__init__.py -> build/lib.linux-x86_64-3.6/horovod/common
creating build/lib.linux-x86_64-3.6/horovod/tensorflow/keras
copying horovod/tensorflow/keras/__init__.py -> build/lib.linux-x86_64-3.6/horovod/tensorflow/keras
copying horovod/tensorflow/keras/callbacks.py -> build/lib.linux-x86_64-3.6/horovod/tensorflow/keras
creating build/lib.linux-x86_64-3.6/horovod/torch/mpi_lib
copying horovod/torch/mpi_lib/__init__.py -> build/lib.linux-x86_64-3.6/horovod/torch/mpi_lib
creating build/lib.linux-x86_64-3.6/horovod/torch/mpi_lib_impl
copying horovod/torch/mpi_lib_impl/__init__.py -> build/lib.linux-x86_64-3.6/horovod/torch/mpi_lib_impl
warning: build_py: byte-compiling is disabled, skipping.
running build_ext
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -std=c++11 -fPIC -O2 -Wall -mf16c -mavx -I/usr/local/include/python3.6m -c build/temp.linux-x86_64-3.6/test_compile/test_cpp_flags.cc -o build/temp.linux-x86_64-3.6/test_compile/test_cpp_flags.o
gcc -pthread -shared build/temp.linux-x86_64-3.6/test_compile/test_cpp_flags.o -o build/temp.linux-x86_64-3.6/test_compile/test_cpp_flags.so
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/usr/local/include/python3.6m -c build/temp.linux-x86_64-3.6/test_compile/test_link_flags.cc -o build/temp.linux-x86_64-3.6/test_compile/test_link_flags.o
gcc -pthread -shared -Wl,--version-script=horovod.lds build/temp.linux-x86_64-3.6/test_compile/test_link_flags.o -o build/temp.linux-x86_64-3.6/test_compile/test_link_flags.so
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -std=c++11 -fPIC -O2 -Wall -mf16c -mavx -I/usr/local/cuda/include -I/usr/local/include/python3.6m -c build/temp.linux-x86_64-3.6/test_compile/test_cuda.cc -o build/temp.linux-x86_64-3.6/test_compile/test_cuda.o
gcc -pthread -shared build/temp.linux-x86_64-3.6/test_compile/test_cuda.o -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -lcudart -o build/temp.linux-x86_64-3.6/test_compile/test_cuda.so
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -std=c++11 -fPIC -O2 -Wall -mf16c -mavx -I/usr/local/cuda/include -I/usr/local/include/python3.6m -c build/temp.linux-x86_64-3.6/test_compile/test_nccl.cc -o build/temp.linux-x86_64-3.6/test_compile/test_nccl.o
gcc -pthread -shared build/temp.linux-x86_64-3.6/test_compile/test_nccl.o -L/usr/local/cuda/lib -L/usr/local/cuda/lib64 -lnccl_static -o build/temp.linux-x86_64-3.6/test_compile/test_nccl.so
----------------------------------------
Command "/usr/local/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-0x3phzja/horovod/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-dm5gilyr/install-record.txt --single-version-externally-managed --compile" failed with error code -4 in /tmp/pip-install-0x3phzja/horovod/
You are using pip version 19.0.3, however version 19.1.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Removing intermediate container 6a964b96d01b
The command '/bin/sh -c ldconfig /usr/local/cuda-10.0/targets/x86_64-linux/lib/stubs && HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_WITH_TENSORFLOW=1 pip install --no-cache-dir horovod==0.15.2 && ldconfig' returned a non-zero code: 1
```

**Comment by alsrgv** [comment]
@armandmcqueen, I suspect this issue is caused by CPU architecture used by DockerHub build machines.  I suspect it's not modern, and TensorFlow/PyTorch typically require SSE or AVX instructions which it may not have, which would cause it to crash with SIGILL (illegal instruction, code -4).  I don't know of a good workaround, other than using more modern servers / VMs to build the docker image.

**Comment by armandmcqueen** [comment]
Ah that would make sense, thanks
